compiler/nimconf.nim

```diff
@@ -248,11 +248,16 @@ proc loadConfigs*(cfg: RelativeFile; cache: IdentCache; conf: ConfigRef; idgen:
     if readConfigFile(configPath, cache, conf):
       configFiles.add(configPath)
 
-  template runNimScriptIfExists(path: AbsoluteFile) =
+  template runNimScriptIfExists(path: AbsoluteFile, isMain = false) =
     let p = path # eval once
-    if fileExists(p):
+    var s: PLLStream
+    if isMain and conf.command == "e":
```

Comment:
Wait what? This has no place in ``nimconf.nim``.

Reply:
`conf.command == "e"` is to fix https://github.com/nim-lang/Nim/issues/15731 (+ an analog error for the new `--eval`), and the `isMain` part is to avoid evaluating `cmdInput` for each config.nims processed.

Reply:
Bugfix or not, it's not nimconf's business to do special logic for ``conf.command == "e"``, this should be done in in upper layer.

Reply:
fixed